cytosnake/helpers/datapaths.py

```diff
@@ -0,0 +1,164 @@
+"""
+module: datapath.py
+
+This module contains fuctions that build stirng path that are snakemake compatible.
+"""
+
+import pathlib
+from typing import Optional
+
+import cytosnake.utils.cyto_paths as cyto_paths
+from cytosnake.guards.path_guards import is_valid_path
+from cytosnake.utils.config_utils import load_general_configs, load_meta_path_configs
+
+# constant wildcard name
+WILDCARD_BASENAME = "{basename}"
+
+# loading in config files
+META_PATH_CONFIGS = load_meta_path_configs()
+GENERAL_CONFIGS = load_general_configs()
+
+
+# -----------------------
+# input paths from configs
+# -----------------------
+def get_barcodes() -> str:
+    """Obtains path to the bardcode file within data folder
+
+    Returns
+    -------
+    str
+        return string with barcode name wildcard
+
+    """
+    # Barcodes are optional. If not added, set to "None"
+    try:
+        barcode_path = META_PATH_CONFIGS["project_dir"]["data_directory_contents"][
+            "barcode"
+        ]
+    except KeyError:
+        barcode_path = None
+
+    return barcode_path
+
+
+def get_metadata_dir() -> str:
+    """Obtains path to the metadata folder with in data folder
+
+    Returns
+    -------
+    str
+        returns string that contains metadata wildcard
+    """
+
+    try:
+        metadata_dir = META_PATH_CONFIGS["project_dir"]["data_directory_contents"][
+            "metadata"
+        ]
+    except KeyError as e:
```

Comment:
Depending on utility, consider how tools like [`jsonschema`](https://github.com/python-jsonschema/jsonschema) might be used with the validation of configuration files.

Reply:
Yes, I would like to develop a schema at some point. Given how much rapid change the configurations files have gone through, I think we do not have a finalized config structure to the point where a schema is necessary. Once we have a robust configuration structure and less frequent changes, then developing a schema is something that should be implemented. Thanks for the suggestion, will add this into the issues!

Reply:
Created issue #55